File: instrumentation/android-log/library/src/main/java/io/opentelemetry/instrumentation/library/log/AndroidLogInstrumentation.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.library.log;

import androidx.annotation.NonNull;
import com.google.auto.service.AutoService;
import io.opentelemetry.android.instrumentation.AndroidInstrumentation;
import io.opentelemetry.android.instrumentation.InstallationContext;
import io.opentelemetry.instrumentation.library.log.internal.LogRecordBuilderCreator;

@AutoService(AndroidInstrumentation.class)
public class AndroidLogInstrumentation implements AndroidInstrumentation {
    public static final String INSTRUMENTATION_NAME = "android-log";

    @Override
    public void install(@NonNull InstallationContext ctx) {
        LogRecordBuilderCreator.configure(ctx);
    }

    @NonNull
    @Override
    public String getName() {
        return INSTRUMENTATION_NAME;
    }
}

File: instrumentation/android-log/library/src/main/java/io/opentelemetry/instrumentation/library/log/AndroidLogInstrumentation.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.library.log

import android.content.Context
import com.google.auto.service.AutoService
import io.opentelemetry.android.OpenTelemetryRum
import io.opentelemetry.android.instrumentation.AndroidInstrumentation
import io.opentelemetry.instrumentation.library.log.internal.LogRecordBuilderCreator.configure

@AutoService(AndroidInstrumentation::class)
class AndroidLogInstrumentation : AndroidInstrumentation {
    override fun install(context: Context, openTelemetryRum: OpenTelemetryRum) {
        configure(openTelemetryRum.openTelemetry)
    }

    override val name: String = "android-log"
}
